### ISSUE TITLE
No 3d view

### ISSUE THREAD

**Issue by JoePfeiffer** [openrocket/openrocket#1251]
We've got one user who is reporting he isn't seeing anything (not even one of the infamous opengl error messages) in the rocket figure window when he switches to 3D.  Since it's only affecting one user (that I know of) I expect there's something in his configuration, but I don't know enough about Windows to be able to make a guess.  He sent me a debug log, which I am attaching.
[no-3d-debug.txt](https://github.com/openrocket/openrocket/files/8287665/no-3d-debug.txt)

**Comment by hcraigmiller** [comment]
See also Issue #864.

**Comment by RemingtonHolder** [comment]
From looking at the log from #864 it looks as if this is an OpenGL issue. Do the used OpenGL libraries differ from v15.03 & the current unstable branch?

**Comment by SiboVG** [comment]
> From looking at the log from #864 it looks as if this is an OpenGL issue. Do the used OpenGL libraries differ from v15.03 & the current unstable branch?

Yes, 15.03 uses javax.media.opengl, but after the 15.03 release, javax.media.opengl was officially ported to org.jogamp.jogl, which OR uses today.

**Comment by RemingtonHolder** [comment]
> > From looking at the log from #864 it looks as if this is an OpenGL issue. Do the used OpenGL libraries differ from v15.03 & the current unstable branch?
> 
> 
> 
> Yes, 15.03 uses javax.media.opengl, but after the 15.03 release, javax.media.opengl was officially ported to org.jogamp.jogl, which OR uses today. 3D functionality with JOGL has been an issue for some time and is something that will likely be addressed after the next official OR release.

This is the best explanation of why the issue is occurring then. At the same time I see no easy solution to this unless you want to dig through the library looking for diffs.

**Comment by SiboVG** [comment]
There's no real use in comparing the diffs in the libs. Since JOGL updates have been almost non-existent since 2015 and compatibility-issues are starting to pile up, we are considering porting 3D functionality to another library. But again, this is something that will be discussed more in detail after the release.

**Comment by JoePfeiffer** [comment]
> From looking at the log from #864 it looks as if this is an OpenGL issue. Do the used OpenGL libraries differ from v15.03 & the current unstable branch?

You're right, it almost has to be an OpenGL issue, and more specifically something in the interaction between the libraries and the Windows drivers on those machines (since it's only been reported on exactly two Windows machines). I don't know how to figure it out from there until somebody suffers from the issue who has the expertise to debug it.

**Comment by hcraigmiller** [comment]
I had an interesting interaction with a TRF user with a Lenovo system. The user could open OR, but could not get the edit boxes to work. The user was utilizing the Windows installer and **_did not have Java installed_**. At one point, the user installed Java 8, with no change, the user could still open OR, but not an edit box. However, after uninstalling Java 8 and, _**after installing Java 11**_, OR began functioning without anomalous behavior.

Just an FYI.

**Comment by SiboVG** [comment]
> I had an interesting interaction with a TRF user with a Lenovo system. The user could open OR, but could not get the edit boxes to work. The user was utilizing the Windows installer and **_did not have Java installed_**. At one point, the user installed Java 8, with no change, the user could still open OR, but not an edit box. However, after uninstalling Java 8 and, _**after installing Java 11**_, OR began functioning without anomalous behavior.
> 
> Just an FYI.

Sounds like something fit to put on a FAQ-page on the website (along with some other common fixes like 'use off-screen rendering')...

**Comment by hcraigmiller** [comment]
Done...
[2022 Beta After-Installation Check ->Check Settings](http://wiki.openrocket.info/Downloading_%26_Installing)
[Problems running OpenRocket](http://wiki.openrocket.info/FAQ#When_will_OpenRocket_run_on_a_newer_version_of_Java.3F)

**Comment by JoePfeiffer** [comment]
> I had an interesting interaction with a TRF user with a Lenovo system. The user could open OR, but could not get the edit boxes to work. The user was utilizing the Windows installer and **_did not have Java installed_**. At one point, the user installed Java 8, with no change, the user could still open OR, but not an edit box. However, after uninstalling Java 8 and, _**after installing Java 11**_, OR began functioning without anomalous behavior.
> 
> Just an FYI.

I've got absolutely no idea how that can be possible....

**Comment by hcraigmiller** [comment]
I don’t know either… I was having him send me screenshots at each step so I could have him try the current unstable… but, before I could send him the .jar, he said it was working. Strange.

**Comment by Rick-S-B5** [comment]
FWIW  I'm having the same issues with 3D views and with the Photo Studio. I have java 11 installed. This is a Win 7 pro running on a HP Pavilion desktop.  I'm getting an "Uncaught Exception" and a Lockup that I can get out of with an "End Task" in Windows Task Manager.  Also, please forgive me I got in here kind of by accident, I'm not a Coder.  Now to find my way out and keep looking else ware for a solution. Thanks guys

**Comment by JoePfeiffer** [comment]
FWIW I just had reason to try to run 24.12 and to compile and run current unstable on VirtualBox Windows hosted on Debian Linux and found exactly this behavior. Toggling offscreen rendering made no difference.
